Sources/xcodeproj/Objects/Project/PBXObject.swift

```diff
@@ -13,6 +13,9 @@ public class PBXObject: Hashable, Decodable, Equatable, AutoEquatable {
     /// The object reference in the project that contains it.
     let reference: PBXObjectReference
 
+    /// Used to differentiate this object from other equatable ones for the purpose of reference generation
+    public var identifier: String?
```

Comment:
What about using the name that you suggested, `referenceContext`? Furthermore, I'd make the variable internal.